src/Fieldset/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import newId from '../utils/newId';
+import styles from './Fieldset.scss';
+import ValidationMessage from '../ValidationMessage/index';
+import Variant from '../utils/constants';
+
+const inputProps = {
+  variantIconDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  children: PropTypes.node,
+  className: PropTypes.arrayOf(PropTypes.string),
+  id: PropTypes.string,
+  isValid: PropTypes.bool,
+  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  invalidMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  variant: PropTypes.shape({
+    status: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  variantIconDescription: '',
+  children: null,
+  className: [],
+  id: '',
+  isValid: true,
+  invalidMessage: '',
+  variant: {
+    status: Variant.status.INFO,
+  },
+};
+
+class Fieldset extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const id = props.id || newId('fieldset');
+    this.state = {
```

Comment:
style nit - consider doing

```javascript
this.state = { id };
```

and save two lines.

or

```javascript
this.state = { id: props.id || newId('fieldset') };
```

and save three.